modules/build/src/main/scala/scala/build/bsp/BspImpl.scala

```diff
@@ -149,9 +151,18 @@ final class BspImpl(
       buildChangedTest
     )
 
+    if (actionableDiagnostics.getOrElse(false))
+      logActionableDiagnostics(options0Test.orElse(options0Main), persistentLogger)
+
     PreBuildProject(mainScope, testScope, persistentLogger.diagnostics)
   }
 
+  private def logActionableDiagnostics(projectOptions: BuildOptions, logger: Logger): Unit = {
```

Comment:
NIT This method is duplicated in this PR (the other instance is in `Build.scala`)
I wonder if this logic could become an extension for `BuildOptions`?